### PR TITLE
refactor(Core/Grids): Ported cmangos/mangos-wotlk@ea99457

### PR DESCRIPTION
**Pull request by Kitzunu**
(cherry picked from commit https://github.com/TrinityCore/TrinityCore/commit/d6201e5dbb7fa7ca8b47ffa6c0d30fa38dceada5)
Co-Authored-By: Shauren <shauren.trinity@gmail.com>
Co-Authored-By: SilverIce <slifeleaf@gmail.com>

Needed as part of https://github.com/azerothcore/azerothcore-wotlk/pull/6109

## Changes Proposed:
-  Ported cmangos/mangos-wotlk@ea99457

## Tests Performed:

- Builds 

## How to Test the Changes:

Play around in-game, feel like if everything is like before

For example use some commands that have changes here, like respawn

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
I'd also cite the original authors, SilveIce and VladimirMangos https://github.com/cmangos/mangos-wotlk/commit/ea99457e50790acde8928aa0f3dc4a9c096b4a8c

**Comment by Kitzunu** [comment]
Added Silverice as he was the author for Mangos

File: src/server/game/Grids/Cells/CellImpl.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license, you may redistribute it and/or modify it under version 2 of the License, or (at your option), any later version.
- * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+ * Copyright (C) 2008-2016 acoreCore <http://www.acorecore.org/>
  * Copyright (C) 2005-2009 MaNGOS <http://getmangos.com/>
```
Comment (Winfidonarleyan) [inline src/server/game/Grids/Cells/CellImpl.h:4]: Rofl

Comment (Kitzunu) [reply src/server/game/Grids/Cells/CellImpl.h:4]: Ah nice catch XD

**Comment by Winfidonarleyan** [comment]
- Part #4302

File: src/server/scripts/Spells/spell_generic.cpp [hunk 2]
```diff
@@ -1721,7 +1721,9 @@ class spell_gen_cannibalize : public SpellScriptLoader
             // search for nearby enemy corpse in range
             acore::AnyDeadUnitSpellTargetInRangeCheck check(caster, max_range, GetSpellInfo(), TARGET_CHECK_CORPSE);
             acore::WorldObjectSearcher<acore::AnyDeadUnitSpellTargetInRangeCheck> searcher(caster, result, check);
-            caster->GetMap()->VisitFirstFound(caster->m_positionX, caster->m_positionY, max_range, searcher);
+            Cell::VisitWorldObjects(caster, searcher, max_range);
+            if (!result)
```
Comment (Winfidonarleyan) [inline src/server/scripts/Spells/spell_generic.cpp:1725]: @FrancescoBorzi He's not going to be happy.

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 3]
```diff
@@ -979,7 +979,9 @@ class spell_hun_pet_carrion_feeder : public SpellScriptLoader
             // search for nearby enemy corpse in range
             acore::AnyDeadUnitSpellTargetInRangeCheck check(caster, max_range, GetSpellInfo(), TARGET_CHECK_ENEMY);
             acore::WorldObjectSearcher<acore::AnyDeadUnitSpellTargetInRangeCheck> searcher(caster, result, check);
-            caster->GetMap()->VisitFirstFound(caster->m_positionX, caster->m_positionY, max_range, searcher);
+            Cell::VisitWorldObjects(caster, searcher, max_range);
+            if (!result)
```
Comment (Winfidonarleyan) [inline src/server/scripts/Spells/spell_hunter.cpp:983]: again

**Review by Winfidonarleyan — APPROVED** [review]

File: src/server/scripts/Spells/spell_generic.cpp (hunk 2 above)
Comment (Kitzunu) [reply src/server/scripts/Spells/spell_generic.cpp:1725]: yikes, will fix all of them

Comment (Winfidonarleyan) [reply src/server/scripts/Spells/spell_generic.cpp:1725]: :D

Comment (Kitzunu) [reply src/server/scripts/Spells/spell_generic.cpp:1725]: He didnt see :o

**Comment by Kitzunu** [comment]
https://github.com/ElunaLuaEngine/Eluna/pull/368 Eluna

File: src/server/game/Grids/Cells/CellImpl.h [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license, you may redistribute it and/or modify it under version 2 of the License, or (at your option), any later version.
- * Copyright (C) 2008-2016 acoreCore <http://www.acorecore.org/>
+ * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
```
Comment (FrancescoBorzi) [inline src/server/game/Grids/Cells/CellImpl.h]: lol acorecore.org

**Comment by Kitzunu** [comment]
Who's the smartass who wanted to rename all the acore namespaces D: That was 44 conflicts D:::

**Comment by tkn963** [comment]
Ran around doing quests and such in several different regions with no issues
Tested the Carrion Feeder hunter pet talent many times with no issues
Was told to check the spell Cannibalize, found it only used by certain creatures in Dragonblight - if this is correct, I also tested it many times with no issues
Tried several commands, the one I tested the most was .respawn but encountered no issues with any command I tested no matter the situation

**Pull request merged**